File: LeetCode/3656-minimum-number-of-operations-to-make-elements-in-array-distinct/minimum-number-of-operations-to-make-elements-in-array-distinct.cpp
class Solution {
public:
    int minimumOperations(vector<int>& nums) {
        unordered_map<int, int> um;
        for(auto it: nums){
            um[it]++;
        }

        int sz = nums.size();
        if(sz == um.size())
            return 0;
            
        int op = 0;
        int i = 0;
        while(true){
            for(int j = 0; j < 3 && i < nums.size(); j++){
                um[nums[i]]--;
                if(um[nums[i]] == 0){
                    um.erase(nums[i]);
                }
                i++, sz--;
            }
            if(um.size() == sz){
                break;
            }
            op++;
        }
        return op+1;
    }
};

File: LeetCode/3656-minimum-number-of-operations-to-make-elements-in-array-distinct/minimum-number-of-operations-to-make-elements-in-array-distinct.go
func minimumOperations(nums []int) int {
    mp := make(map[int]int)
    sz := len(nums)
    for i := 0; i < sz; i++ {
        mp[nums[i]]++;
    } 

    if sz == len(mp) {
        return 0;
    }

    op := 0;
    i := 0;

    for{
        for j := 0; j < 3 && i < len(nums); j++ {
            mp[nums[i]]--
            if mp[nums[i]] == 0 {
                delete(mp, nums[i])
            }
            sz--
            i++
        }
        if len(mp) == sz {
            break
        }
        op++
    }
    return op + 1;
}